### PR TITLE
Blog App: Add devise

### PR DESCRIPTION
**Pull request by arnoldnekemiah**
### Welcome to this PR
In this PR I configured Devise to enable Authentication;
Removed current_user method in ApplicationController, devise provided us with one.
[Install and setup devise](https://github.com/heartcombo/devise#getting-started).

-   Can register a new user.
-   User logs in with a combination of email and password.
-   Hashed password is stored in the database.
-   Asks for confirmation of email.
-   Can reset password.
- Added basic styling to make it look good

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (zieeco) [inline app/models/user.rb]: - Hey, let's also make sure that `confirmable` is enabled to meet the project requirement.

File: db/migrate/20231109142322_add_devise_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      # t.datetime :locked_at
+
+
+      # Uncomment below if timestamps were not included in your original model.
+      # t.timestamps null: false
+    end
+
+    add_index :users, :email,                unique: true
+    add_index :users, :reset_password_token, unique: true
+    # add_index :users, :confirmation_token,   unique: true
```
Comment (zieeco) [inline db/migrate/20231109142322_add_devise_to_users.rb]: - After enabling the `confirmable` in the devise setup for the `user` model, let's also enable this to make it work correctly.

File: app/views/devise/sessions/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+  </div>
+
```
Comment (zieeco) [inline app/views/devise/sessions/new.html.erb]: - Hey, from the wireframe, the inputs and button take up the full width, please, kindly improve on this.

*wireframe*
image


*your design*
image

Comment (zieeco) [reply app/views/devise/sessions/new.html.erb]: - Kindly make the same changes for your sign_up, the design pattern follows the same wireframe.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,





Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Devise fairly setup. ✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @arnoldnekemiah 

Your project is complete! Great job on implementing the changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- wireframe well followed
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**